### PR TITLE
[SYCLomatic] Replace the vec_size of device RNG engines from 4 to 1 in the migrated code and update the related helper functions

### PR DESCRIPTION
**Pull request by zhiweij1**

### CODE REVIEW

**Comment by zhiweij1** [comment]
mcg59 engine does not have ctor like `mcg59 (std::uint32_t seed, std::initializer_list<std::uint64_t> offset)`

**Comment by zhiweij1** [comment]
@andreyfe1 @aelizaro , could you help review the MKL API usage part in this PR?

**Comment by andreyfe1** [comment]
Looks correct to me. So, we keep <4> and <1> support for philox and mrg, correct?

**Comment by zhiweij1** [comment]
> Looks correct to me. So, we keep <4> and <1> support for philox and mrg, correct?

Right. We need keep backward compatibility for those public APIs.

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/rng_utils.hpp.inc [hunk 1]
```diff
@@ -57,14 +57,24 @@ namespace device {
...
+/// oneapi::mkl::rng::device::mrg32k3a<1> or
 /// oneapi::mkl::rng::device::mrg32k3a<4> or
+/// oneapi::mkl::rng::device::philox4x32x10<1> or
 /// oneapi::mkl::rng::device::philox4x32x10<4>.
 template <typename engine_t> class rng_generator {
   static_assert(
-      std::is_same_v<engine_t, oneapi::mkl::rng::device::mrg32k3a<4>> ||
+      std::is_same_v<engine_t, oneapi::mkl::rng::device::mrg32k3a<1>> ||
+          std::is_same_v<engine_t, oneapi::mkl::rng::device::mrg32k3a<4>> ||
+          std::is_same_v<engine_t,
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/rng_utils.hpp.inc]: can use std::disjunction

**Review by andreyfe1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**